### PR TITLE
Fetchers into the backend

### PR DESCRIPTION
**Pull request by JakeSteinebronn**
# Describe your changes
Move the fetchers into the backend. Every fetchable object has
- async fetch()
- buildFetcher()
- static buildFetcherByDocRefId
# Issue ticket number and link
#182

### CODE REVIEW

**Review by Henry-Graves — COMMENTED** [review]
lgtm but I want to point your attention to the syntax in my first comment. Also taking sends off User is some nice cleanup

File: types/post.ts [hunk 1]
```diff
@@ -30,6 +30,24 @@ import {
...
+  likes: User[];
+  imageContentUrls: string[];
+
+  forum: Forum | undefined;
+  videoContent:
+    | {
+        videoUrl: string;
+        thumbnailUrl: string;
+      }
+    | undefined;
```
[the remark this answers is not in the document]
Comment (tylerhm) [reply types/post.ts:46]: This is via formatter. Extra union symbols are ignored by Typescript, so this just allows for clean left-justification.

[the remark this answers is not in the document]
Comment (tylerhm) [reply types/post.ts:46]: The `?` notation implies that the property could not exist on the object which is slightly different than "has to exist, but it's value might be undefined". I think the latter feels a bit more correct here but feel free to change it if you disagree.

**Comment by AlexRutledge1** [comment]
I'm liking what I'm seeing, is there any chance that we can also get a `fetchByID` function for users and routes? Web uses search params for navigating to routes and user profiles so being able to call the fetch function without having the class yet would be very useful.

**Comment by AlexRutledge1** [comment]
> I'm liking what I'm seeing, is there any chance that we can also get a `fetchByID` function for users and routes? Web uses search params for navigating to routes and user profiles so being able to call the fetch function without having the class yet would be very useful.

Ignore me, lmao. Just saw its already implemented. This is why you read each line before commenting.

**Review by AlexRutledge1 — APPROVED** [review]
Looks great

File: types/route.ts [hunk 2]
```diff
@@ -96,6 +96,29 @@ export interface EditRouteArgs {
...
+  status: RouteStatus;
+  description: string;
+  thumbnailUrl: string;
+
+  setter?: User;
+  setterRawName?: string;
+  rope?: number;
+  timestamp?: Date; // Defined if active or archived
+  color?: string;
+  naturalRules?: NaturalRules;
```
Comment (DanielW48) [inline types/route.ts:114]: can we add a stars field to this fetcher? there is now a backend getter for it

**Pull request merged**